### PR TITLE
`Assessment`: Improve user experience during repository export

### LINKED ISSUE #7158: Download repo automatically checks anonymize repository
### Describe the bug

When downloading a student's repo during assessment the `Anonymize repository` option gets automatically checked

### To Reproduce

1. Assess a student's programming submission as tutor/instructor
2. Click on 'Download Repo'
3. Click on 'Export'
4. Then `Anonymize repository` gets automatically checked

### Expected behavior

Judging from the code this is expected behavour but should not be displayed (double-blind assessment)

### Screenshots

When clicking download repo **HERE**
Screenshot 2023-09-08 at 10 54 09

The **Anonymize repository** option gets automatically checked on export
Screenshot 2023-09-08 at 10 54 18

### Which version of Artemis are you seeing the problem on?

6.4.3

### What browsers are you seeing the problem on?

Chrome, Safari

### Additional context

@b-fein found this bug

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
This PR closes #7158. The "Anonymize Repository" Flag has been checked once the export is submitted, regardless of what the user checked the box before. This is bad UX. This PR makes the UI with the used flags consistent and simplifies the conditions by allowing instructors to change and see the "anonymize" flag, whereas tutors don't see it and always have an anonymized repository

### Description
This PR removes the force-changed export options from the exportRepos() method to make the UI consistent with the actual behavior of the export. Also the condition is simplified (and slightly changed), so that Instructors are allowed to change the anonymize flag and tutors are not.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Tutor
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis as a Student
2. Submit a solution for the Programming exercise
3. Log in to Artemis as an instructor
4. Change The "Due Date" to now
5. Assess the submition
6. Click "Export Repository"
7. Verify that you see the checkbox for "Anonymize Repository"
8. Verify that the checkbox stays checked/unchecked depending on what you chose once submitted and that the downloaded Repository is anonymous/not anonymous depending on your choice.

9. Log out and log in as Tutor and verify that you don't see the "Anonymize Repository" Flag
10. Export the repository and see that it is anonymized.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
Instructor:
image

Tutor:
image

### CODE REVIEW

**Review by FelixTJDietrich — DISMISSED** [review]
Code looks good and works as expected on ts5.

Correctly disabled checkbox: ✅
Screenshot 2023-10-05 at 12 57 45

Export dialog is unchanged and anonymization checkbox is unchecked: ✅
Screenshot 2023-10-05 at 12 58 15

**Review by jakubriegel — CHANGES_REQUESTED** [review]
Code looks ok, but on ts5 the checkbox is active:
image

**Comment by FelixTJDietrich** [comment]
> Code looks ok, but on ts5 the checkbox is active: image

Which checkbox are you referring to? `Anonymize repository` should be unchecked here by default and checked when downloading a repo during assessment.

**Review by jakubriegel — APPROVED** [review]
After finding the right export button works as it should:
image

**Review by Strohgelaender — COMMENTED** [review]
I don't like the solution of using `!!participationIdList?.length` as a condition. This might work for now, but could breaks easily if the export component gets added to another places.

Can you not just check the access rights of the current user? If the user is only tutor, the anomymize repository option should always be active and disabled.

File: src/main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-dialog.component.html [hunk 1]
```diff
@@ -75,7 +75,13 @@ <h4 class="modal-title" jhiTranslate="entity.exportRepos.title">Confirm export o
         </div>
         <div *ngIf="!singleParticipantMode" class="checkbox">
             <label class="control-label">
-                <input class="form-check-input" type="checkbox" name="addParticipantName" [(ngModel)]="this.repositoryExportOptions.addParticipantName" />
+                <input
+                    class="form-check-input"
+                    type="checkbox"
+                    name="addParticipantName"
+                    [(ngModel)]="this.repositoryExportOptions.addParticipantName"
+                    [disabled]="!!participationIdList?.length"
```
Comment (milljoniaer) [inline src/main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-dialog.component.html:83]: I have forgotten to also disable this checkbox if `!!participationIdList?.length` applies to stick to the logic before.

**Review by RY997 — APPROVED** [review]
Tested on ts5, worked well.

**Review by jakubriegel — APPROVED** [review]
Code ok after update

**Review by Strohgelaender — DISMISSED** [review]
lgtm now

**Review by Strohgelaender — APPROVED** [review]
re-approve code

**Review by laurenzfb — APPROVED** [review]
Tested in testing session, lgmt!

**Review by aplr — APPROVED** [review]
Tested in testing session, works well, aside from the follow-up. LGMT

**Review by laadvo — APPROVED** [review]
Code looks good

**Review by FelixTJDietrich — APPROVED** [review]
Code changes LGTM

**Pull request merged**